supervisor/security/module.py

```diff
@@ -0,0 +1,169 @@
+"""Fetch last versions from webserver."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from ..const import (
+    ATTR_CONTENT_TRUST,
+    ATTR_FORCE_SECURITY,
+    ATTR_PWNED,
+    FILE_HASSIO_SECURITY,
+)
+from ..coresys import CoreSys, CoreSysAttributes
+from ..exceptions import (
+    CodeNotaryError,
+    CodeNotaryUntrusted,
+    PwnedError,
+    SecurityJobError,
+)
+from ..jobs.decorator import Job, JobCondition, JobExecutionLimit
+from ..resolution.const import ContextType, IssueType
+from ..utils.codenotary import cas_validate
+from ..utils.common import FileConfiguration
+from ..utils.pwned import check_pwned_password
+from ..validate import SCHEMA_SECURITY_CONFIG
+from .const import ContentTrustResult, IntegrityResult
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+class Security(FileConfiguration, CoreSysAttributes):
+    """Handle Security properties."""
+
+    def __init__(self, coresys: CoreSys):
+        """Initialize updater."""
+        super().__init__(FILE_HASSIO_SECURITY, SCHEMA_SECURITY_CONFIG)
+        self.coresys = coresys
+
+    @property
+    def content_trust(self) -> bool:
+        """Return if content trust is enabled/disabled."""
+        return self._data[ATTR_CONTENT_TRUST]
+
+    @content_trust.setter
+    def content_trust(self, value: bool) -> None:
+        """Set content trust is enabled/disabled."""
+        self._data[ATTR_CONTENT_TRUST] = value
```

Comment:
Re-enabling this should run a full integrity check. Since we only check content on installs now we may have missed some.

Reply:
Possibly although if there's an issue with CAS again they might temporarily disable it to just stop seeing errors. Running it on re-enable could be helpful although I suppose they can just do that themselves.